Samples do not work

### Environment
Node version: 8.11.1
Npm version: 6.0.1
Tsc version: 3.4.1
OS and version: Win10 1803
azure-devops-node-api version: Git cloned master branch

### Issue Description
I can't get the samples to run.

### Expected behaviour
Samples run successfully

### Actual behaviour
Errors out

### Steps to reproduce
1. git clone
2. cd into directory
3. npm install
4. npm run samples

### Logs

> C:\repos\azure-devops-node-api-2> npm run samples
> 
> > azure-devops-node-api@7.3.0 samples C:\repos\azure-devops-node-api-2
> > node make.js samples
> 
> > tsc --outDir C:\repos\azure-devops-node-api-2\_build
> C:\repos\azure-devops-node-api-2\samples C:\repos\azure-devops-node-api-2
> > npm install ../_build
> npm WARN vsts-node-api-samples@1.0.0 No repository field.
> 
> + azure-devops-node-api@7.3.0
> added 4 packages from 8 contributors in 2.236s
> [+] no known vulnerabilities found [6 packages audited]
> 
> C:\repos\azure-devops-node-api-2
> > tsc -p samples
> samples/workItemTracking.ts(70,53): error TS2551: Property 'getWorkItemTypeFields' does not exist on type 'IWorkItemTrackingApi'. Did you mean 'getWorkItemTypeColors'?
> C:\repos\azure-devops-node-api-2\node_modules\shelljs\src\common.js:417
>       if (config.fatal) throw e;
>                         ^
> 
> Error: exec: internal error
>     at Object.error (C:\repos\azure-devops-node-api-2\node_modules\shelljs\src\common.js:128:27)
>     at _exec (C:\repos\azure-devops-node-api-2\node_modules\shelljs\src\exec.js:292:12)
>     at C:\repos\azure-devops-node-api-2\node_modules\shelljs\src\common.js:350:23
>     at run (C:\repos\azure-devops-node-api-2\make.js:13:14)
>     at Function.target.samples (C:\repos\azure-devops-node-api-2\make.js:67:5)
>     at Object.global.target.(anonymous function) [as samples] (C:\repos\azure-devops-node-api-2\node_modules\shelljs\make.js:36:40)
>     at C:\repos\azure-devops-node-api-2\node_modules\shelljs\make.js:48:27
>     at Array.forEach (<anonymous>)
>     at Timeout._onTimeout (C:\repos\azure-devops-node-api-2\node_modules\shelljs\make.js:46:10)
>     at ontimeout (timers.js:482:11)
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! azure-devops-node-api@7.3.0 samples: `node make.js samples`
> npm ERR! Exit status 1

To fix, I changed

```
witApi.getWorkItemTypeFields
```

to

```
witApi.getWorkItemTypeFieldsWithReferences
```

and reran `npm run samples` but I get a permissioning error:

> > tsc -p samples
> C:\repos\azure-devops-node-api-2\samples C:\repos\azure-devops-node-api-2
> > node run.js
> Hello XXXXX XXXXX
> 
> > Creating example project
> Cleaning up from last run
> Nothing to clean up
> (node:12364) UnhandledPromiseRejectionWarning: Error: Failed request: (401)
>     at RestClient.<anonymous> (C:\repos\azure-devops-node-api-2\_build\node_modules\typed-rest-client\RestClient.js:200:31)
>     at Generator.next (<anonymous>)
>     at fulfilled (C:\repos\azure-devops-node-api-2\_build\node_modules\typed-rest-client\RestClient.js:6:58)
>     at <anonymous>
>     at process._tickCallback (internal/process/next_tick.js:188:7)
> (node:12364) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
> (node:12364) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> C:\repos\azure-devops-node-api-2

Note that it shows I'm authenticated with the "Hello" message, but then I get a 401 anyway. I'm using a PAT with full access.

Is it user error? Or is something broken?

--- a/samples/common.ts
+++ b/samples/common.ts
@@ -10,8 +10,8 @@ function getEnv(name: string): string {
     return val;
 }
 
-export async function getWebApi(): Promise<vm.WebApi> {
-    let serverUrl = getEnv("API_URL");
+export async function getWebApi(serverUrl?: string): Promise<vm.WebApi> {
+    serverUrl = serverUrl || getEnv("API_URL");
     return await this.getApi(serverUrl);
 }
 

--- a/samples/profile.ts
+++ b/samples/profile.ts
@@ -5,7 +5,11 @@ import * as ProfileApi from "azure-devops-node-api/ProfileApi";
 import * as ProfileInterfaces from "azure-devops-node-api/interfaces/ProfileInterfaces";
 
 export async function run(projectId: string) {
-    const webApi: nodeApi.WebApi = await common.getWebApi();
+    let serverUrl = process.env["API_URL"];
+    // Profile api can't be hit at the org level, has to be hit at the deployment level, so url should be structured like set API_URL=https://vssps.dev.azure.com/{orgName}
+    serverUrl = serverUrl.replace('://dev.azure.com', '://vssps.dev.azure.com');
+    console.log('serverUrl', serverUrl);
+    const webApi: nodeApi.WebApi = await common.getWebApi(serverUrl);
     const profileApiObject: ProfileApi.IProfileApi = await webApi.getProfileApi();
     common.banner("Profile Samples");
 
@@ -13,19 +17,11 @@ export async function run(projectId: string) {
     const createProfileContext: ProfileInterfaces.CreateProfileContext = {cIData: null,
                                                                           contactWithOffers: false,
                                                                           countryName: "US",
-                                                                          displayName: null,
+                                                                          displayName: "displayedName",
                                                                           emailAddress: "sample@microsoft.com",
                                                                           hasAccount: false,
                                                                           language: "english",
                                                                           phoneNumber: "123456890"};
     let profile: ProfileInterfaces.Profile = await profileApiObject.createProfile(createProfileContext, false);
     console.log("Profile created for", profile.coreAttributes.DisplayName.value);
-
-    common.heading("Get the avatar");
-    const avatar: ProfileInterfaces.Avatar = await profileApiObject.getAvatar(profile.id);
-    console.log("Avatar value:", avatar.value);
-
-    common.heading("Get region information");
-    const regions: ProfileInterfaces.ProfileRegions = await profileApiObject.getRegions();
-    console.log(`There are ${regions.regions.length} regions found including regions like ${regions.regions[0].name}`);
 }

